### PR TITLE
Add a lua function to check for prerequisites

### PR DESCRIPTION
**Pull request by abcdefg30**

### CODE REVIEW

**Comment by chrisforbes** [comment]
:+1:

File: OpenRA.Mods.Common/Scripting/Properties/PlayerProperties.cs [hunk 1]
```diff
@@ -74,5 +74,15 @@ public Actor[] GetActorsByType(string type)
...
 		}
+
+		[Desc("Check if the player has this prerequisite available.")]
+		public bool HasPrerequisiteAvailable(string type)
+		{
+			var tt = Player.PlayerActor.TraitOrDefault<TechTree>();
+			if (tt == null)
+				throw new LuaException("Missing TechTree trait on player {0}!".F(Player));
+
+			return tt.HasPrerequisites(type.OfType<string>());
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Scripting/Properties/PlayerProperties.cs]: Hmm, this doesn't seem to work. `TechTree.HasPrerequisites` returns `true` for a prerequisite the player doesn't actually have.

Testcase:
```diff
diff --git a/mods/cnc/maps/gdi01/gdi01.lua b/mods/cnc/maps/gdi01/gdi01.lua
index 1855f74..55ad272 100644
--- a/mods/cnc/maps/gdi01/gdi01.lua
+++ b/mods/cnc/maps/gdi01/gdi01.lua
@@ -97,11 +97,17 @@ WorldLoaded = function()
        Trigger.AfterDelay(DateTime.Seconds(60), function() Reinforce(VehicleReinforcements) end)
 end
 
+done = false
 Tick = function()
        if enemy.HasNoRequiredUnits() then
                player.MarkCompletedObjective(secureAreaObjective)
        end
 
+       if not done and player.HasPrerequisiteAvailable("nuke") then
+               Media.DisplayMessage("Power plant built.")
+               done = true
+       end
+
        if DateTime.GameTime > DateTime.Seconds(5) and player.HasNoRequiredUnits() then
                player.MarkFailedObjective(beachheadObjective)
                player.MarkFailedObjective(secureAreaObjective)
```

File: OpenRA.Mods.Common/Scripting/Properties/PlayerProperties.cs [hunk 2]
```diff
@@ -74,5 +74,15 @@ public Actor[] GetActorsByType(string type)
 
 			return result.ToArray();
 		}
+
+		[Desc("Check if the player has this prerequisites available.")]
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Scripting/Properties/PlayerProperties.cs]: "these" now :)

**Comment by obrakmann** [comment]
Works as expected now. :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/8f42e7e0ff5f7d2a07bc3bfecb2fb99710c0fc5d)

**Pull request merged**